### ISSUE TITLE
USWDS - Bug: Time Picker fails pa11y accessibility scans due to hidden text input

### ISSUE THREAD

**Issue by ajanickiv** [uswds/uswds#4954]
### Describe the bug

The hidden text input in the Time Picker does not pass accessibility scans with pa11y.

Error: This textinput element does not have a name available to an accessibility API. Valid names are: label element, title undefined, aria-label undefined, aria-labelledby undefined.
[htmlcs: WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.InputText.Name](https://squizlabs.github.io/HTML_CodeSniffer/Standards/WCAG2/#sc4_1_2)
`<input type="text" class="usa-input" aria-describedby="appointment-time-label appointment-time-hint" style="display: none;">`

Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrapped around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
[htmlcs: WCAG2AA.Principle1.Guideline1_3.1_3_1.F68](https://squizlabs.github.io/HTML_CodeSniffer/Standards/WCAG2/#sc1_3_1)
`<input type="text" class="usa-input" aria-describedby="appointment-time-label appointment-time-hint" style="display: none;">`
![image](https://user-images.githubusercontent.com/16595489/191313562-77b88e34-dd95-4385-978d-a354a9d7f92d.png)

### Steps to reproduce the bug

1. Go to https://designsystem.digital.gov/components/time-picker/
2. Run an accessibility scan with pa11y

### Expected Behavior

The pa11y scan to pass when running a scan of the time picker.

### Related code

In usa-time-picker/index.js the transformTimePicker function adds a select input and moves attributes like id and aria-label from the text input to the select. It then hides the text input. After that, enhanceComboBox is called which adds another text input and hides the select. Maybe instead of hiding the initial text input it should be removed?

### System setup

- USWDS Version: 3.0.2

### Additional context

The combo box does not seem to have the hidden text input. Maybe it is not necessary in the time picker?

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by amyleadem** [comment]
Confirming that these errors appear when testing https://designsystem.digital.gov/components/time-picker/ on pa11y v6.2.3

**Comment by mejiaj** [comment]
@ajanickiv @amyleadem I might be misunderstanding the problem, but w3 states in [using-aria section](https://www.w3.org/TR/using-aria/#issue-container-generatedID-5)

>If an interactive element is hidden using display:none or visibility:hidden (either on the element itself, or any of the element's ancestors), it won't be focusable, and it will also be removed from the [accessibility tree](https://www.w3.org/TR/accname-aam-1.1/#dfn-accessibility-tree). This makes the addition of aria-hidden="true" or explicitly setting tabindex="-1" unnecessary.

Shouldn't that make this error a false positive with pa11y?

**Comment by amyleadem** [comment]
@mejiaj 
According to [this issue](https://github.com/pa11y/pa11y/issues/652) in the pa11y repo (answered just this morning), it looks to be tied to an issue with the HTML CodeSniffer runner.

The error does not appear when using the axe runner in pa11y on the component preview page:

Returns errors:
```
pa11y https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/iframe.html?args=&id=components-form-inputs-time-picker--time-picker&viewMode=story
```

Returns no errors:
```
pa11y --runner axe https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/iframe.html?args=&id=components-form-inputs-time-picker--time-picker&viewMode=story
```
![image](https://user-images.githubusercontent.com/93996430/197232361-4c56c8ff-8e24-4bdb-9a8b-fd57f4a4a4d3.png)

**Comment by mejiaj** [comment]
Okay, thanks for checking on that @amyleadem!

**Comment by ajanickiv** [comment]
Thanks for the info! If switching to the axe runner is not an option, it is possible to ignore these two issues.  Here is my pa11y-ci ignore block
`"ignore": [
    "WCAG2AA.Principle1.Guideline1_3.1_3_1.F68",
    "WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.InputText.Name"
]`
